### DISCUSSION TITLE
EVCC config, where to get SOC?

### DISCUSSION THREAD

**Discussion by swa72** [wimaha/TeslaBleHttpProxy#37] [General]
https://github.com/wimaha/TeslaBleHttpProxy#setup-evcc refers to an easier setup than previous versions. The description should probably read 

```
vehicles:
  - name: model3
      type: template
     [...]
```
Where does EVCC now get the SoC for the car?

**Comment by dglaude** [comment]
I get my soc from TeslaMate and it is fetch from mqtt. (an alternative might be TeslaLogger)
I just made sure to expose the mqtt port in the `docker-compose.yml`.
While it is working, I have evcc complaining about out of date value, it is an annoyance I could not fix yet.

Another way is to use the official evcc tesla template (that require a support token from github/evcc), and use the proxy feature to have TeslaBleHttpProxy do the BLE trick, while using the cloud API for less frequent call (such as soc): https://docs.evcc.io/en/docs/devices/vehicles#tesla

But here is "my" way with TeslaMate via MQTT:
```
  soc:
    source: mqtt
    topic: teslamate/cars/1/battery_level
    timeout: 60s
  range:
    source: mqtt
    topic: teslamate/cars/1/est_battery_range_km
    timeout: 60s
  status:
    source: combined
    plugged:
      source: mqtt
      topic: teslamate/cars/1/plugged_in
      timeout: 60s
    charging:
      source: mqtt
      topic: teslamate/cars/1/charging_state
      timeout: 60s

# mqtt message broker
mqtt:
  broker: 192.168.192.157:1883
```

I am no expert and I am still confused about:
* Those error message I get about range(?).
* There is teslamate support in evcc https://docs.evcc.io/en/docs/devices/vehicles#teslamate but I don't know how to use that in my case.

PS: Right now I gave up on using TeslaBleHttpProxy and I use script executed from evcc and that use the CLI tesla-control software. I also gave up on using Pi4 and moved all the component: evcc, teslamate and tesla-control (and maybe one day I will re-install TeslaBleHttpProxy) on a single NUC.

**Comment by swa72** [comment]
Note sure, if your hint applies in my case. In my case https://github.com/alandtse/tesla pipes all Tesla data into Home Assistant. Same holds for [Tesla Wall Connector](https://www.home-assistant.io/integrations/tesla_wall_connector).
Home assistant in turn publishes SoC via MQTT and I could consume that in EVCC as well. So far so good.

I assume that TeslaBleHttpProxy only sends data and does not receive anything.  So EVCC must get some data (SoC, plugged-in, etc.) from somewhere else. I could provide that data via MQTT.

I'm simply confused which `template:` to use.  The one on https://github.com/wimaha/TeslaBleHttpProxy#setup-evcc for version 0.128.0 does not appear to require any more input.

**Comment by swa72** [comment]
With the config under https://github.com/wimaha/TeslaBleHttpProxy#setup-evcc EVCC did not communicate to the car at all. I had to add the `vin:` line. Now it does communicate with the car. 

```
2024/07/30 11:54:01 DEBU close connection (A)
2024/07/30 17:31:06 INFO received command=charge_stop body=map[]
2024/07/30 17:31:06 INFO connecting to Vehicle ...
2024/07/30 17:31:06 DEBU trying connecting to vehicle attempt=1
2024/07/30 17:31:06 DEBU connecting to vehicle (A)...
2024/07/30 17:31:08 DEBU create vehicle object ...
2024/07/30 17:31:08 DEBU connecting to vehicle (B)...
2024/07/30 17:31:08 DEBU start VCSEC session...
2024/07/30 17:31:09 DEBU car successfully wakeup
2024/07/30 17:31:09 DEBU start Infotainment session...
2024/07/30 17:31:10 INFO connection established
2024/07/30 17:31:10 INFO sending command=charge_stop body=map[]
2024/07/30 17:31:11 INFO successfully executed command=charge_stop body=map[]
2024/07/30 17:31:40 DEBU connection Timeout
2024/07/30 17:31:40 DEBU disconnect vehicle (A)
2024/07/30 17:31:40 DEBU close connection (A)
```
and, to my surprise, once enough sun was available, EVCC did use the commandProxy to trigger a charge. 

```
2024/07/30 17:40:10 INFO received command=set_charging_amps body=map[charging_amps:1]
2024/07/30 17:40:10 INFO connecting to Vehicle ...
2024/07/30 17:40:10 DEBU trying connecting to vehicle attempt=1
2024/07/30 17:40:10 DEBU connecting to vehicle (A)...
2024/07/30 17:40:10 INFO received command=charge_start body=map[]
2024/07/30 17:40:10 DEBU create vehicle object ...
2024/07/30 17:40:10 DEBU connecting to vehicle (B)...
2024/07/30 17:40:10 DEBU start VCSEC session...
2024/07/30 17:40:11 DEBU car successfully wakeup
2024/07/30 17:40:11 DEBU start Infotainment session...
2024/07/30 17:40:12 INFO connection established
2024/07/30 17:40:12 INFO sending command=set_charging_amps body=map[charging_amps:1]
2024/07/30 17:40:13 INFO successfully executed command=set_charging_amps body=map[charging_amps:1]
2024/07/30 17:40:13 INFO sending command=charge_start body=map[]
2024/07/30 17:40:14 INFO successfully executed command=charge_start body=map[]
2024/07/30 17:40:37 INFO received command=set_charging_amps body=map[charging_amps:3]
2024/07/30 17:40:37 INFO sending command=set_charging_amps body=map[charging_amps:3]
2024/07/30 17:40:38 INFO successfully executed command=set_charging_amps body=map[charging_amps:3]
2024/07/30 17:40:42 DEBU connection Timeout
2024/07/30 17:40:42 DEBU disconnect vehicle (A)
2024/07/30 17:40:42 DEBU close connection (A)

```
Bottom line, here is the config that worked for me ...
```
vehicles:
  - name: T1 # no spaces or special characters
    type: template
    template: tesla
    accessToken: XXX
    refreshToken: XXX
    vin: XXX
    commandProxy: http://192.168.178.78:8080
```
What I don't quite get is where EVCC receives SoC and the other stuff from?

**Reply by dglaude** [reply]
You are using the official way to do it, with a valid evcc support token.

evcc used to do everything via the cloud and using tesla key.
Then, about a month ago, Tesla changed the number of API calls one could do per day, making changing the charging Amps not possible anymore.
This is where TeslaBleHttpProxy came in and was integrated via the proxy command.

For those four commands `wake_up`, `charge_start`, `charge_stop` and `set_charging_amps` evcc will ask via TeslaBleHttpProxy, for other request such as soc, it will contact Tesla Cloud. At least that is what I understand.

Maybe this discussion can help understand the history: https://github.com/evcc-io/evcc/discussions/14252

I don't use evcc tesla template, and for totally unknown reason TeslaBleHttpProxy is not reliably working at home (I tested of two Pi and an Intel NUC). So right now I cannot use evcc official solution in partnership with TeslaBleHttpProxy... so I use a custom template that call script for all the charge command, and I use Teslamate for fetching the SoC...

It kind of work, except that I get those 3 warnings/errors very frequently and it pollute the interface:
* `vehicle range: outdated`
* `vehicle soc: outdated`
* `vehicle soc: outdated (ignored by estimator)`

Regards

**Comment by swa72** [comment]
Although I did read the mentioned thread before, I did not understand it - too many different approaches and little to no documentation ;-) Thanks for the elaborate answer that cleared things up for me.
